src/Microsoft.Azure.WebJobs.Host/Loggers/TraceEventExtensions.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.WebJobs.Host.Protocols;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    internal static class TraceEventExtensions
+    {
+        private const string FunctionDescriptorKey = "MS_FunctionDescriptor";
+        private const string FunctionInvocationIdKey = "MS_FunctionInvocationId";
+        private const string HostIdKey = "MS_HostId";
+
+        public static void AddFunctionInstanceDetails(this TraceEvent traceEvent, Guid hostId, FunctionDescriptor descriptor, Guid functionId)
+        {
+            traceEvent.Properties[HostIdKey] = hostId;
```

Comment:
HostID was explicitly asked for in #774. Don't you have to construct and register your tracer with the Config before you have a Host id? That's what I thought -- but I can double-check if there's any way to do that.